src/api-v1.js

```diff
@@ -667,11 +702,19 @@ api.declare({
   }
 
   let loadedAmiSet = await this.AmiSet.load({id});
+  let invalidAmis = await this.AmiSet.validate(this, loadedAmiSet);
```

Comment:
Let's validate before we load... We know that if we don't validate that we can return immediately.  Chances are also good that we're more likely to fail validation than loading, so not too much risk in wasted effort.